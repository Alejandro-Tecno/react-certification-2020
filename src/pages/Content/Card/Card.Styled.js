import styled from "styled-components";
import { Link } from "react-router-dom";


export const StyledLink = styled(Link)`
  text-decoration: none;

  &:focus,
  &:hover,
  &:visited,
  &:link,
  &:active {
    text-decoration: none;
  }
`;

export const StyledCardHolder = styled.div`
  position: relative;
  &:hover button {
    display: flex !important;
  }
  button {
    cursor: pointer;
    display: none;
    z-index: 99;
    color: white;
    background: #d60e0e;
    position: absolute;
    top: 5%;
    right: 5%;
    padding: 5px !important;
    font-size: 1.5rem;
    border-radius: 5px;
    box-shadow: 0 0 20px black;
    justify-content: space-around;
    align-items: center;
  }
`;

export const StyledCard = styled.div`
  display: flex;
  flex-direction: column;
  width: 350px;
  align-items: center;
  position: relative;
  border: 1px solid #3a3a53;
  border-radius: 5px;
  margin: 10px 10px;
  text-decoration: none;
  transition: all 0.3s ease-in-out;

  &:hover {
    transform: scale(1.02);

  }

  .video-image {
    height: 190px;
    width: 350px;
    border-radius: 5px 5px 0px 0px;
    background-size: cover;
    background-repeat: no-repeat;
    background-position: center;
    display: flex;

    img {
      display: block;
      width: 100%;
      height: auto;
      object-fit: cover;
      border-radius: 5px 5px 0px 0px;
    }
  }
  .video-content {
    padding: 0rem 1rem;
    height: 150px;
    h2 {
      font-size: 1.2rem;
      text-decoration: none;
      overflow: hidden;
      text-overflow: ellipsis;
      height: 3rem;
    }
    p {
      font-size: 0.8rem;
      text-decoration: none;
      overflow: hidden;
      text-overflow: ellipsis;
    }
  }
`;
