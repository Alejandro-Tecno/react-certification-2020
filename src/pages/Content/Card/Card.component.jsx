import React from "react";
import styled from "styled-components";
import { Link } from "react-router-dom";

function Card({ image, title, description, id }) {
  return (
    <StyledLink to={`/video/${id}`}>
      <StyledCard data-testid="card_div">
        <div className="video-image">
          <img data-testid="card_img" src={image} alt="" />
        </div>
        <div className="video-content">
          <h2 data-testid="card_h2">
            {title.replace(/&#39;/, "'").replace(/&amp;/, "&")}
          </h2>
          <p data-testid="card_p">{description}</p>
        </div>
      </StyledCard>
    </StyledLink>
  );
}

const StyledLink = styled(Link)`
  text-decoration: none;

  &:focus,
  &:hover,
  &:visited,
  &:link,
  &:active {
    text-decoration: none;
  }
`;

const StyledCard = styled.div`
  display: flex;
  flex-direction: column;
  width: 350px;
  align-items: center;
  background: #ffffff;
  position: relative;
  border: 1px solid #e7e5e5;
  border-radius: 5px;
  margin: 10px 10px;
  text-decoration: none;
  transition: all 0.3s ease-in-out;
  box-shadow: 12px 12px 24px 0 rgba(0, 0, 0, 0.016),
    -12px -12px 24px 0 rgba(255, 255, 255, 0.1);
  &:hover {
    transform: scale(1.02);
    box-shadow: 12px 12px 24px 0 rgba(0, 0, 0, 0.116),
      -12px -12px 24px 0 rgba(255, 255, 255, 0.5);
  }

  .video-image {
    height: 190px;
    width: 350px;
    border-radius: 5px 5px 0px 0px;
    background-size: cover;
    background-repeat: no-repeat;
    background-position: center;
    display: flex;
    img {
      display: block;
      width: 100%;
      height: auto;
      object-fit: cover;
    }
  }
  .video-content {
    padding: 0rem 1rem;
    height: 150px;
    h2 {
      font-size: 1.2rem;
      text-decoration: none;
      color: black;
      overflow: hidden;
      text-overflow: ellipsis;
      height: 3rem;
    }
    p {
      font-size: 0.8rem;
      color: #403d4e;
      text-decoration: none;
      overflow: hidden;
      text-overflow: ellipsis;
    }
  }
`;

export default Card;
