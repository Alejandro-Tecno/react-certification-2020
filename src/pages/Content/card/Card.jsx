import React from 'react';
import styled from 'styled-components';

function Card({ image, title, description }) {
  const StyledCard = styled.div`
    display: flex;
    flex-direction: column;
    width: 350px;
    align-items:center;
    background: #ffffff;
    position:relative;
    border: 2px solid #c4c2c2;
    border-radius: 5px;
    margin: 10px 0px;

    .video-image {
      background: url(${image});
      height: 150px;
    width: 350px;
    border-radius: 5px 5px 0px 0px;
    background-size: cover;
    background-repeat: no-repeat;
    background-position: center;
    }
    .video-content{
        padding: 0rem 1rem;
        h2{
            font-size:1.2rem;
        }
        p{
            font-size: .8rem;
            color: #403d4e;
        }
    }
  `;

  return (
    <StyledCard>
      <div className="video-image"> </div>
      <div className="video-content">
        <h2>{title}</h2>
        <p>{description}</p>
      </div>
    </StyledCard>
  );
}

export default Card;
