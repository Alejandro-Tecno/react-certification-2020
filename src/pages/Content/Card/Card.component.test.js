import React from "react";
import Card from "./";
import reactDom from "react-dom";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

describe("Card creator", () => {
  it("Renders correctly", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Card
            image="https://i.ytimg.com/vi/zClI9OjgKXM/hqdefault.jpg"
            title="Video&#39;1"
            description="testDescription"
          />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renders the H2 and p tags correctly", () => {
    // Initializes variables
    const { getByTestId, getByRole } = render(
      <BrowserRouter>
        <Card
          image="https://i.ytimg.com/vi/zClI9OjgKXM/hqdefault.jpg"
          title="Video&#39;1"
          description="testDescription"
        />
      </BrowserRouter>
    );
    // Check if the elements rendered are the expected
    expect(getByTestId("video_description")).toBeInTheDocument();
    expect(getByTestId("video_description")).toHaveTextContent(
      "testDescription"
    );
    expect(getByTestId("card_h2")).toHaveTextContent("Video'1");
    expect(getByRole("img")).toHaveProperty(
      "src",
      "https://i.ytimg.com/vi/zClI9OjgKXM/hqdefault.jpg"
    );
  });
});
