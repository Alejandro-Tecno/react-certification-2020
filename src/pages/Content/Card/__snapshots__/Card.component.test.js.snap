// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card creator Renders correctly 1`] = `
<a
  className="sc-bdfBwQ czeCVk"
  href="/video/undefined"
  onClick={[Function]}
>
  <div
    className="sc-gsTCUz dKMgDE"
    data-testid="card_div"
  >
    <div
      className="video-image"
      data-testid="video-image"
    >
      <img
        alt=""
        data-testid="card_img"
        src="https://i.ytimg.com/vi/zClI9OjgKXM/hqdefault.jpg"
      />
    </div>
    <div
      className="video-content"
    >
      <h2
        data-testid="card_h2"
      >
        Video'1
      </h2>
      <p
        data-testid="video_description"
      >
        testDescription
      </p>
    </div>
  </div>
</a>
`;
